### PR TITLE
ODC-7531: Added the ApprovalTask List Tabbed Page

### PR DESCRIPTION
**Pull request by Lucifergene**
**Fixes**: 
https://issues.redhat.com/browse/ODC-7531

**Depends on**:
https://issues.redhat.com/browse/ODC-7519

**Screen shots / Gifs for design review**: 
image

image

Demo:

https://github.com/openshift/console/assets/47265560/8dfb9ab8-1c2e-48b0-aa59-11ac22dbff51

**Unit test coverage report**: 
Fixed broken tests. New tests will be added as part of another story.

**Test setup:**
1. Install Pipelines operator and Install the ApprovalTask Controller following this [Doc](https://github.com/openshift-pipelines/manual-approval-gate/blob/e1cf38bb54fa379c0efe271aec26ab7fa3fe4227/docs/DEVELOPEMENT.md).
2. Execute the following gist: https://gist.github.com/Lucifergene/f491504d7eda8a97e86a132029a9bc7b

**Browser conformance**: 

- [x] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by invincibleJai** [comment]
This PR is dependent on https://github.com/openshift/console/pull/13613

**Comment by MariaLeonova** [comment]
@Lucifergene loos good! One question regarding the design in the Approvers column: it looks like the task was approved by 1 out of 4 approvers, and yet it's overall status is 'Approved'. How can that be?

**Comment by Lucifergene** [comment]
> @Lucifergene looks good! One question regarding the design in the Approvers column: it looks like the task was approved by 1 out of 4 approvers, and yet it's overall status is 'Approved'. How can that be?

@MariaLeonova, in the ApprovalTask resource YAML, we can specify the number of approvals required for the job. In this case, it was set to 1. Therefore, even on getting just one approval, it got approved.

File: frontend/packages/pipelines-plugin/src/components/tasks/list-page/approval-tasks/ApprovalHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { sortable } from '@patternfly/react-table';
+import i18next from 'i18next';
+import { tableColumnClasses } from './approval-table';
+
+const ApprovalHeader = () => {
+  return [
+    {
+      title: i18next.t('pipelines-plugin~PipelineRun Name'),
```
Comment (vikram-raj) [inline frontend/packages/pipelines-plugin/src/components/tasks/list-page/approval-tasks/ApprovalHeader.tsx:8]:
```suggestion
      title: i18next.t('pipelines-plugin~PipelineRun name'),
```

File: frontend/packages/pipelines-plugin/src/components/tasks/list-page/approval-tasks/ApprovalHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      props: { className: tableColumnClasses.plrName },
+      id: 'plrName',
+    },
+    {
+      title: i18next.t('pipelines-plugin~Approvers'),
+      props: { className: tableColumnClasses.approvers },
+      id: 'approvers',
+    },
+    {
+      title: i18next.t('pipelines-plugin~Task Name'),
```
Comment (vikram-raj) [inline frontend/packages/pipelines-plugin/src/components/tasks/list-page/approval-tasks/ApprovalHeader.tsx:20]:
```suggestion
      title: i18next.t('pipelines-plugin~Task name'),
```

File: frontend/packages/pipelines-plugin/src/components/tasks/list-page/approval-tasks/ApprovalHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      props: { className: tableColumnClasses.taskName },
+      id: 'taskName',
+    },
+    {
+      title: i18next.t('pipelines-plugin~Current Status'),
```
Comment (vikram-raj) [inline frontend/packages/pipelines-plugin/src/components/tasks/list-page/approval-tasks/ApprovalHeader.tsx:25]:
```suggestion
      title: i18next.t('pipelines-plugin~Current status'),
```

File: frontend/packages/pipelines-plugin/locales/en/pipelines-plugin.json [hunk 4]
```diff
@@ -437,11 +438,22 @@
...
+  "PipelineRun Name": "PipelineRun Name",
+  "Approvers": "Approvers",
+  "Task Name": "Task Name",
+  "Current Status": "Current Status",
+  "Approve": "Approve",
+  "Reject": "Reject",
+  "Approval status": "Approval status",
+  "ApprovalTasks": "ApprovalTasks",
+  "An error occurred. Please try again": "An error occurred. Please try again",
+  "<0>{type === 'approve'\n          ? 'Are you sure you want to approve'\n          : 'Please provide a reason for not approving'} <2></2><3>{name}</3> in <6></6><7></7><8>{pipelineRunName}</8>?</0>": "<0>{type === 'approve'\n          ? 'Are you sure you want to approve'\n          : 'Please provide a reason for not approving'} <2></2><3>{name}</3> in <6></6><7></7><8>{pipelineRunName}</8>?</0>",
```
Comment (vikram-raj) [inline frontend/packages/pipelines-plugin/locales/en/pipelines-plugin.json:450]: It doesn't look fine. The translation string contains conditions as well. Can you fix it.

File: frontend/packages/pipelines-plugin/src/components/tasks/list-page/approval-tasks/ApprovalTasksListPage.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        const status = pipelineApprovalFilterReducer(obj, pipelineRuns, customRuns);
+        return !filterValue.selected?.length || (status && filterValue.selected.includes(status));
+      },
+    },
+  ];
+
+  return (
+    <>
+      <Helmet>
+        <title>{t('pipelines-plugin~ApprovalTasks')}</title>
```
Comment (vikram-raj) [inline frontend/packages/pipelines-plugin/src/components/tasks/list-page/approval-tasks/ApprovalTasksListPage.tsx:62]:
```suggestion
        <title>{t('pipelines-plugin~Approval tasks')}</title>
```

**Review by vikram-raj — COMMENTED** [review]
Verified it. This feature works as expected. But there is some improvement suggested by  UX and PM which can be done in the followup PR. As PR owner is on PTO. 

/lgtm

**Comment by sanketpathak** [comment]
Verified happy paths with admin and non-admin users. As edge cases will be dealt in another pr

https://github.com/openshift/console/assets/10252230/f3332061-2224-4e8c-a34c-8c3d7de5adac

/label qe-approved

**Pull request merged**